### DISCUSSION TITLE
Stochastic inputs to the system

### DISCUSSION THREAD

**Discussion by Hofsmo** [NREL-SIIP/PowerSimulationsDynamics.jl#272] [Q&A]
Hi,
I have been working a bit with system identification in power system. In these studies it is important that the power system is excited by an external signal. I typically do this by modelling random load changes as a Wiener-process. Another important concept is that of process noise, which I sometimes model as filtered white noise added to for instance the rotational speed of a synchronous machine. 

I found this simple tutorial showing how to add stochasticity to an ODEProblem https://matbesancon.xyz/post/2017-12-14-diffeq-julia/ . I was thinking if maybe it is possible to first create the problem using your package and then later add stochasticity by creating an SDEProblem as in the tutorial. I did several unsuccessful attempts of achieving this, so I was wondering if maybe there are better ways of doing this.

If you have any hints on how I can add stochastic inputs to simulations using your package, I would be very happy.

Cheers Sigurd

**Comment by jd-lara** [comment]
hi @Hofsmo we can add this as a request but I think we aren't well positioned at the moment to make this a feature. If you have ideas, please paste them here or open a PR and we start exploring the use of PSID with stochastic inputs.

**Reply by Hofsmo** [reply]
Hi @jd-lara, sorry for the late reply. I regularly checked for replies, but since the thread was marked as _unanswered_ I thought there was no answer. 

It would be nice to have it as a request :), but I understand if it not prioritised at the moment. I have done some tests using the [DifferentialEquations.jl](https://diffeq.sciml.ai/stable/) library, and have managed to create stochastic problems using problems constructed by your library. However, I have not managed to run it yet. If I manage to create something that works, I will create a PR :) .
